source/common/stats/heap_stat_data.h

```diff
@@ -23,9 +23,10 @@ namespace Stats {
  * This structure is an alternate backing store for both CounterImpl and GaugeImpl. It is designed
  * so that it can be allocated efficiently from the heap on demand.
  */
-struct HeapStatData {
+struct HeapStatData : public InlineStorage {
```

Comment:
It's a little magical to me that `symbol_storage_` is what is inline, given that we have a layer of indirection via another struct/object declaration. I'm not sure if there is any way to make this better but maybe a comment somewhere?